diesel/src/pg/expression/functions.rs

```diff
@@ -748,3 +748,34 @@ define_sql_function! {
     /// ```
     fn array_append<Arr: ArrayOrNullableArray<Inner=T> + SingleValue, T: SingleValue>(a: Arr, e: T) -> Array<T>;
 }
+
+#[cfg(feature = "postgres_backend")]
+define_sql_function! {
+    /// Returns a text representation of the array's dimensions
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main(){
+    /// #    run_test().unwrap();
+    /// # }
+    /// # fn run_test()->QueryResult<()>{
+    /// #   use diesel::dsl::array_dims;
+    /// #   use diesel::sql_types::{Nullable,Array,Integer};
+    /// #   let connection = &mut establish_connection();
+    ///
+    /// let dims = diesel::select(array_dims::<Array<_>,_,_>(vec![1,2]))
```

Comment:
```suggestion
    /// let dims = diesel::select(array_dims::<Array<Integer>,_,_>(vec![1,2]))
```
The error says that rust don't know with implementation use, this `i32` could be `Int4`, `Int8` or something else, to fix you need to specify the type here. The `array_append` doesn't need this because it Array inner type is the same of the second argument, and the second argument is specified

Reply:
Thanks for the help!